honeybee_energy/run.py

```diff
@@ -36,24 +42,83 @@ def run_idf_windows(idf_file_path, epw_file_path):
     # check and prepare the input files
     directory = prepare_idf_for_simulation(idf_file_path, epw_file_path)
 
+    # generate the object expansion string if requested
+    if expand_objects:
+        exp_path = os.path.join(folders.energyplus_path, 'ExpandObjects')
+        exp_str = '{}\nif exist expanded.idf MOVE expanded.idf in.idf\n'.format(exp_path)
+    else:
+        exp_str = ''
+
     # write a batch file
-    expand_path = os.path.join(folders.energyplus_path, 'ExpandObjects')
     run_path = os.path.join(folders.energyplus_path, 'EnergyPlus')
     working_drive = directory[:2]
-    batch = '{}\ncd {}\n{}\nif exist expanded.idf MOVE expanded.idf in.idf\n{}'.format(
-        working_drive, directory, expand_path, run_path)
-    write_to_file_by_name(directory, 'in.bat', batch, True)
+    batch = '{}\ncd {}\n{}{}'.format(working_drive, directory, exp_str, run_path)
+    batch_file = os.path.join(directory, 'in.bat')
+    write_to_file(batch_file, batch, True)
 
     # run the batch file
-    os.system(os.path.join(directory, 'in.bat'))
+    os.system(batch_file)
 
     # output the simulation files
-    return _output_files_from_directory(directory)
+    return _output_energyplus_files(directory)
+
+
+def run_idf_unix(idf_file_path, epw_file_path, expand_objects=True):
```

Comment:
Since these methods run the simulation I think we should have a single run_idf which senses the platform and does it correctly based on the operating system.

Reply:
Good suggestion. I will implement this.

Reply:
Done.